packages/editor/styles/styles.css

```diff
@@ -118,6 +118,11 @@
  margin-bottom: 0px;
 }
 
+.ProseMirror .taskitem-content-wrapper p {
+  margin-top: 0em !important;
+  margin-bottom: 0px !important;
+ }
```

Comment:
This will only fix the margin issue for task items. It won't fix it for other places where paragraph is used. You need to fix the base paragraph margins so they only apply to the top most paragraphs and paragraphs in callout and blockquotes